packages/mds-cache/index.ts

```diff
@@ -353,40 +353,37 @@ async function readDevices(device_ids: UUID[]) {
 }
 
 async function readDeviceStatus(device_id: UUID) {
-  let event: VehicleEvent
-  let device: Device
-  const cachedInfo = []
-  try {
-    event = await readEvent(device_id)
-    cachedInfo.push(event)
-  } catch (err) {
-    if (err.name !== 'NotFoundError') {
-      throw err
-    }
-    log.info('Missing vehicle event', 'device_id', device_id)
-  }
-  try {
-    device = await readDevice(device_id)
-    cachedInfo.push(device)
-  } catch (err) {
-    if (err.name !== 'NotFoundError') {
-      throw err
-    }
-    log.info('Missing vehicle device', 'device_id', device_id)
-  }
-
-  const deviceStatusMap: { [device_id: string]: CachedItem | {} } = {}
-  cachedInfo.map(item => {
-    deviceStatusMap[item.device_id] = deviceStatusMap[item.device_id] || {}
-    Object.assign(deviceStatusMap[item.device_id], item)
-  })
-  const statuses = Object.values(deviceStatusMap)
-  const statusWithTelemetry = statuses.find((status: any) => status.telemetry)
-  if (statusWithTelemetry === undefined) {
-    log.info('Missing vehicle telemetry', 'device_id', device_id)
-    return statuses[0]
-  }
-  return statusWithTelemetry
+  let ret: {} | null = null
+  const promises = [readEvent(device_id), readDevice(device_id)]
+  // Catch all NotFoundErrors
+  await Promise.all(
```

Comment:
In general, I try to avoid using `.catch` and `.then`, and instead try to use `try/catch` and `await`.